crates/bevy_math/src/curve/interval.rs

```diff
@@ -0,0 +1,333 @@
+    #[inline]
+    pub fn has_finite_start(self) -> bool {
+        self.start.is_finite()
+    }
+
+    /// Returns `true` if this interval has a finite end.
+    #[inline]
+    pub fn has_finite_end(self) -> bool {
+        self.end.is_finite()
+    }
+
+    /// Returns `true` if `item` is contained in this interval.
+    #[inline]
+    pub fn contains(self, item: f32) -> bool {
+        (self.start..=self.end).contains(&item)
+    }
+
+    /// Returns `true` if the other interval is contained in this interval.
+    ///
+    /// This is non-strict: each interval will contain itself.
+    #[inline]
+    pub fn contains_interval(self, other: Self) -> bool {
+        self.start <= other.start && self.end >= other.end
+    }
+
+    /// Clamp the given `value` to lie within this interval.
+    #[inline]
+    pub fn clamp(self, value: f32) -> f32 {
+        value.clamp(self.start, self.end)
+    }
+
+    /// Get an iterator over equally-spaced points from this interval in increasing order.
+    /// If `points` is 1, the start of this interval is returned. If `points` is 0, an empty
+    /// iterator is returned. An error is returned if the interval is unbounded.
+    #[inline]
+    pub fn spaced_points(
+        self,
+        points: usize,
+    ) -> Result<impl Iterator<Item = f32>, InfiniteIntervalError> {
+        if !self.is_finite() {
+            return Err(InfiniteIntervalError);
+        }
+        if points < 2 {
+            // If `points` is 1, this is `Some(self.start)` as an iterator, and if `points` is 0,
+            // then this is `None` as an iterator. This is written this way to avoid having to
+            // introduce a ternary disjunction of iterators.
+            let iter = (points == 1).then_some(self.start).into_iter();
+            return Ok(Either::Left(iter));
+        }
+        let step = self.length() / (points - 1) as f32;
+        let iter = (0..points).map(move |x| self.start + x as f32 * step);
+        Ok(Either::Right(iter))
+    }
+
+    /// Get the linear function which maps this interval onto the `other` one. Returns an error if either
+    /// interval is infinite.
+    #[inline]
+    pub(super) fn linear_map_to(
+        self,
+        other: Self,
+    ) -> Result<impl Fn(f32) -> f32, InfiniteIntervalError> {
+        if !self.is_finite() || !other.is_finite() {
+            return Err(InfiniteIntervalError);
+        }
+        let scale = other.length() / self.length();
+        Ok(move |x| (x - self.start) * scale + other.start)
+    }
+}
+
+impl TryFrom<RangeInclusive<f32>> for Interval {
+    type Error = InvalidIntervalError;
+    fn try_from(range: RangeInclusive<f32>) -> Result<Self, Self::Error> {
+        Interval::new(*range.start(), *range.end())
+    }
+}
+
+/// Create an [`Interval`] with a given `start` and `end`. Alias of [`Interval::new`].
+#[inline]
+pub fn interval(start: f32, end: f32) -> Result<Interval, InvalidIntervalError> {
```

Comment:
IMO `unwrap` should be used in contexts where you know it will not panic. We use it far to much in contexts where it 'usually wont panic' and as a result bevy is far too easy to crash.